src/apps/opening-hours-editor/helper.ts

```diff
@@ -172,3 +173,26 @@ export const isOpeningHourWeeklyRepetition = (
     DplOpeningHoursListGET200ItemRepetitionType.weekly
   );
 };
+
+// This used for cypress testing in in cms where we test the complete flow of
+// opening hours between the editor (Admin) and the calendar (Frontend)
+export const getInitialDateFromUrl = (): Date | null => {
+  const query = new URLSearchParams(window.location.search);
+  const initialDateString = query.get("initialDate");
+
+  if (!initialDateString) {
+    return null;
+  }
+
+  const date = new Date(initialDateString);
+  if (isDate(date)) {
+    return date;
```

Comment:
This might be me worrying too much. I expect that you have tested this but when I reason about it I wonder if it works.

According to [the Lodash docs](https://lodash.com/docs/4.17.15#isDate) `isDate()` checks if value is classified as a Date object.

[According to MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#return_value) `new Date()` returns a new Date object but the return value of different methods will signal whether it is a date object.

[According to a Lodash issue `isDate()` will not return false if the value is not a date](https://github.com/lodash/lodash/issues/5040).